sys/include/usb/cdc.h

```diff
@@ -0,0 +1,245 @@
+extern "C" {
+#endif
+
+#define USB_TYPE_DESCRIPTOR_CDC      0x24   /**< USB CDC type descriptor*/
+#define USB_CDC_VERSION_BCD          0x0120 /**< USB CDC version in BCD */
+
+/**
+ * @name USB CDC subclass types
+ * @anchor usb_cdc_subtype
+ * @{
+ */
+#define USB_CDC_SUBCLASS_NONE          0x00 /**< No subclass */
+#define USB_CDC_SUBCLASS_DLCM          0x01 /**< Direct Line Control Model */
+#define USB_CDC_SUBCLASS_ACM           0x02 /**< Abstract Control Model */
+#define USB_CDC_SUBCLASS_TCM           0x03 /**< Telephone Control Model */
+#define USB_CDC_SUBCLASS_MCCM          0x04 /**< Multi-Channel Control Model */
+#define USB_CDC_SUBCLASS_CCM           0x05 /**< CAPI Control Mode */
+#define USB_CDC_SUBCLASS_ENCM          0x06 /**< Eth Networking Control Model */
+#define USB_CDC_SUBCLASS_ANCM          0x07 /**< ATM Networking Control Model */
+#define USB_CDC_SUBCLASS_WHCM          0x08 /**< Wireless Handset Control Model */
+#define USB_CDC_SUBCLASS_DM            0x09 /**< Device Management */
+#define USB_CDC_SUBCLASS_MDLM          0x0A /**< Mobile Direct Line Model */
+#define USB_CDC_SUBCLASS_OBEX          0x0B /**< OBEX */
+#define USB_CDC_SUBCLASS_EEM           0x0C /**< Ethernet Emulation Model */
+#define USB_CDC_SUBCLASS_NCM           0x0D /**< Network Control Model */
+/** @} */
+
+/**
+ * @name USB CDC protocol types
+ * @{
+ */
+#define USB_CDC_PROTOCOL_NONE          0x00 /**< No protocol required */
+#define USB_CDC_PROTOCOL_ITU           0x01 /**< AT Commands: V.250 etc */
+#define USB_CDC_PROTOCOL_PCCA          0x02 /**< AT Commands defined by PCCA-101 */
+#define USB_CDC_PROTOCOL_PCCA_A        0x03 /**< AT Commands defined by PCCA-101 & Annex O */
+#define USB_CDC_PROTOCOL_GSM           0x04 /**< AT Commands defined by GSM 07.07 */
+#define USB_CDC_PROTOCOL_3GPP          0x05 /**< AT Commands defined by 3GPP 27.007 */
+#define USB_CDC_PROTOCOL_CS            0x06 /**< AT Commands defined by TIA for CDMA */
+#define USB_CDC_PROTOCOL_EEM           0x07 /**< Ethernet Emulation Model */
+#define USB_CDC_PROTOCOL_EXT           0xFE /**< External Protocol */
+#define USB_CDC_PROTOCOL_VENDOR        0xFF /**< Vendor-specific */
+/** @} */
+/**
+ * @name USB CDC descriptor subtypes
+ */
+#define USB_CDC_DESCR_SUBTYPE_FUNCTIONAL    0x00 /**< Header functional
+                                                   *  descriptor */
+#define USB_CDC_DESCR_SUBTYPE_CALL_MGMT     0x01 /**< Call management
+                                                      descriptor */
+#define USB_CDC_DESCR_SUBTYPE_ACM           0x02 /**< Abstract control
+                                                      management descriptor */
+#define USB_CDC_DESCR_SUBTYPE_UNION         0x06 /**< Union descriptor */
+#define USB_CDC_DESCR_SUBTYPE_ETH_NET       0x0f /**< Ethernet descriptor */
+/** @} */
+
+/**
+ * @name USB CDC management requests
+ * @{
+ */
+
+/**
+ *USB_CDC_DESCR_SUBTYPE_ACM @brief Set ethernet multicast filter request
```

Comment:
Why USB_CDC_DESCR_SUBTYPE_ACM is here ?

Reply:
Looks like hitting `P` or middle-mouse key while expecting the cursor to be somewhere else :D.

Reply:
Don't know, don't want to know :)